src/server/player_sao.h

```diff
@@ -235,6 +235,7 @@ struct PlayerHPChangeReason
 	enum Type : u8
 	{
 		SET_HP,
+		SET_HP_MAX, // internal type to allow distinguishing hp reset and damage (for effects)
```

Comment:
New enumeration values must be added to the end for matching network serialization. See `clientpackethandler.cpp` for the read-out.